### PR TITLE
RHINENG-26091 Numeric consistency between JSON and CSV responses

### PR DESCRIPTION
**Pull request by saltgen**
## Why do we need this change? :thought_balloon:

Since the numbers were being re-calculated for csv there were some precision mismatches.
Numbers from json response are now being used for CSV.

## Documentation update? :memo:

- [ ] Yes
- [x] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Does this change depend on specific version of Kruize
  - If yes what is the version no:
  - [ ] Is that image available in production or needs deployment?
- [x] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

## Summary by Sourcery

Ensure CSV recommendations reuse the same JSON-derived numeric values to maintain precision consistency between response formats.

Bug Fixes:
- Align CSV numeric fields with their JSON counterparts to avoid precision mismatches in recommendation data.

Enhancements:
- Refactor CSV generation to read from pre-parsed RecommendationsJSON and use consistent float-to-string formatting across outputs.

Tests:
- Add regression test to verify CPU limit values in CSV exactly match those in the JSON for various boundary float64 values.
- Update existing CSV generation tests to construct recommendations via the JSON update path.

### CODE REVIEW

File: internal/api/utils_test.go [hunk 1]
```diff
@@ -198,3 +205,33 @@ func TestInjectStoredRequestVariationPct(t *testing.T) {
...
+// TestJSONvsCSVCPULimitAmount verifies that current_cpu_limit_amount is identical in
+// JSON and CSV for boundary float64 values (e.g. 2.034 = 2.033999... in IEEE 754).
+func TestJSONvsCSVCPULimitAmount(t *testing.T) {
+	for _, amount := range []float64{2.034, 1.0, 0.5, 2.1, 10.999, 0.001, 0.064, 100.123, 0.333, 7.0} {
+		rec := fmt.Sprintf(`{
+			"current": {
+				"limits":   {"cpu": {"amount": %v}},
+				"requests": {"cpu": {}}
+			},
+			"recommendation_terms": {
```
[the remark this answers is not in the document]
Comment (saltgen) [reply internal/api/utils_test.go:218]: The test added in this PR aims to be minimal without losing coverage, we can have such a full fledged test on the IQE side.
Thoughts @Aurobinda55?

[the remark this answers is not in the document]
Comment (Aurobinda55) [reply internal/api/utils_test.go:218]: @saltgen good suggestion .. 
I have  extended the existing test to cover all the terms, including short, medium, and long, for all the  combinations like (e.g., cpu > limit > current, memory > limit > current, cpu > request> current, memory > request > current, etc.)

**Comment by Aurobinda55** [comment]
Looks good to me, added a full-fledged test to cover all the terms and metrics combination in recommendation for both json and csv format.

**Review by Aurobinda55 — APPROVED** [review]
